plugins/woocommerce/src/Blocks/Patterns/PTKClient.php

```diff
@@ -66,4 +127,16 @@ public function fetch_patterns( array $options = array() ) {
 
 		return $decoded_body;
 	}
+
+	/**
+	 * Validate the patterns toolkit patterns.
+	 *
+	 * @param array $patterns The patterns to validate.
+	 * @return bool
+	 */
+	public function is_valid_ptk_patterns( array $patterns ) {
+		$is_pattern_payload_valid = rest_validate_value_from_schema( $patterns, $this->schema );
+
+		return ! is_wp_error( $is_pattern_payload_valid );
```

Comment:
Technically, this class is named `PTKClient`, so it would be reasonable to expect only HTTP client-related code here. However, since this feature is in maintenance mode, it might not be worth creating a dedicated class at this point.